GW2EIEvtcParser/EIData/ProfHelpers/Thief/ThiefHelper.cs

```diff
@@ -24,7 +26,10 @@ internal static class ThiefHelper
             new BuffGiveCastFinder(SoulStoneVenomSkill,SoulStoneVenomEffect),
             new BuffGiveCastFinder(SpiderVenomSkill,SpiderVenomEffect).UsingChecker((evt, combatData, agentData, skillData) => evt.To != evt.By || Math.Abs(evt.AppliedDuration - 24000) < ServerDelayConstant).UsingNotAccurate(true), // same id as leeching venom trait?
             new EffectCastFinder(Pitfall, EffectGUIDs.ThiefPitfallAoE).UsingSrcBaseSpecChecker(Spec.Thief),
-            new EffectCastFinder(ThousandNeedles, EffectGUIDs.ThiefThousandNeedlesAoE1).UsingSrcBaseSpecChecker(Spec.Thief),
+            new BuffLossCastFinder(ThousandNeedlesArmedEffect, ThousandNeedlesArmedEffect),
+            new EffectCastFinder(ThousandNeedles, EffectGUIDs.ThiefThousandNeedlesAoE1)
```

Comment:
I don't see anything solving the 280 ms delay issue.

Reply:
I can add an offset, but the delay can be less than 280ms. I think having the instant cast show up in the log before the actual cast happened is worse than having it show up around the first hit?

Reply:
I don't understand your point, how the instant cast detection could trigger before the actual cast happens? Yes, it is not correct to show the cast after or before it happened, if the timing of the cast is wrong and can only be wrong with given detection than it should be removed.
I don't see how this related to this situation however? You said the colliding effect was the trigger with the other 2 delayed by around 280ms but I see now you are using a buff loss while checking those two effects with a delay, are the buff loss and colliding effect equivalent? If yes, and the buff loss corresponds to the moment the cast fires, I don't see any problem with the proposed implementation.